pkg/controller/statefulset/stateful_set.go

```diff
@@ -422,6 +436,7 @@ func (ssc *StatefulSetController) sync(key string) error {
 		return err
 	}
 	set, err := ssc.setLister.StatefulSets(namespace).Get(name)
+	setCopy := set.DeepCopy()
```

Comment:
I can't see this set is mutated.

Reply:
The set is only used to adopt orphaned controller revisions and for pods listing + adoption by controllerRefManager. So we do not mutate the set here.
Also to sync the satetfulSet we currently deepCoy the set in `#syncStatefulSet()` due to some mutation of the set happening on the `#updateStatefulSet()` method (not sure it is still happening so cc @enisoc @smarterclayton)